synapse/config/federation.py

```diff
@@ -65,5 +65,19 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         self.max_long_retries = federation_config.get("max_long_retries", 10)
         self.max_short_retries = federation_config.get("max_short_retries", 3)
 
+        # Allow for the configuration of the backoff algorithm used
+        # when trying to reach an unavailable destination.
+        # Unlike previous configuration those values applies across
+        # multiple requests and the state of the backoff is stored on DB.
+        self.destination_min_retry_interval_ms = Config.parse_duration(
+            federation_config.get("destination_min_retry_interval", "10m")
+        )
+        self.destination_retry_multiplier = federation_config.get(
+            "destination_retry_multiplier", 2
+        )
+        self.destination_max_retry_interval_ms = Config.parse_duration(
+            federation_config.get("destination_max_retry_interval", "1d")
+        )
```

Comment:
Probably would be good to backoff for a little longer as a day of downtime could easily happen over a weekend before someone bothers to address or just completely dead servers.

For reference, when backfilling a given event, we backoff up to 10 days,

https://github.com/matrix-org/synapse/blob/d53547352063c3e66302f7bdc40d7ac7270e605c/synapse/storage/databases/main/event_federation.py#L77-L90

Reply:
I mean that we might want to try less hard given that a couple days of downtime probably isn't uncommon.

Just being conscious that some we block the client on some federation requests and every time we try to use dead servers, it just increases response time.

Checking each day isn't unreasonable though if you feel strongly about the current tuning.

Reply:
Ok no I thought you meant the reverse since we could also argue that we want a server with a couple of days of downtime to come back as soon as possible. I am fine with a week, let's do so.